### DISCUSSION TITLE
Can you pass a blob to Scene Viewer?

### DISCUSSION THREAD

**Discussion by ZackARC-solutions** [google/model-viewer#2884] [Q&A]
I have tried to pass a blob to the src of model-viewer to load an AR scene on my 3D designer at runtime. This works fine for iOS's quick-look, but scene viewer and webxr both crash when I try load the AR scene. Is it allowed to pass a blob to Scene Viewer? as exporting the glb and hard coding it the src does make the scene viewer work.

**Comment by elalish** [comment]
Pretty sure that doesn't work for Scene Viewer. Can you give some details about the WebXR crash? What do you see in the console? can you share a repro link?

**Reply by ZackARC-solutions** [reply]
Thanks for the reply, WebXR doesn't crash sorry. However, it runs at or below 1 fps. I have a Samsung S20 plus so can't imagine it's hardware limitations. 
Is there a better way to dynamically pass a glb from a three.js scene to Scene Viewer?

**Reply by elalish** [reply]
The 1fps thing in WebXR seems like the much bigger problem. Can you  give a repro link? The rendering performance should be the same between 3D and WebXR modes. Can you give some specifics of your OS and browsers versions as well?

No, Scene Viewer doesn't have that kind of API, which is why we're defaulting to WebXR mode now. Hence it's very important that we get any  problems with WebXR taken care of. Besides, WebXR and SceneViewer have basically the same performance anyway, so the problem would likely exist there too.

**Reply by ZackARC-solutions** [reply]
I'm using:
Google Chrome version: 94.0.4606.85
OS: Android 11; SM-G986B
But I've tested it on a few different Android phones and it has the same problem. I will give you access to the repro.

**Reply by ZackARC-solutions** [reply]
I have sent you a pepro link - it is hosted in arc-solutions.co.uk/test - hit view in AR

**Reply by elalish** [reply]
It appears you have a huge number of draw calls, which will definitely impact performance (on top of ARCore's impact). I'd recommend you collapse the model into fewer meshes and materials.

**Reply by ZackARC-solutions** [reply]
It is strange that iOS can handle it fine. I've tried reducing the model complexity and it has improved slightly, but still is running very slowly. Do you recommend using something like SimplifyModifier? It is almost like it's still rendering the whole scene during the WebXR session. Thank you for your help.

**Reply by elalish** [reply]
ah,  in fact it is rendering the whole scene. We  turn off frustum  culling  (because it  has some animation bugs)  and since MV is nearly always viewing  the entire object anyway. Are you  hoping to  do more of an in-world walk around  kind of  WebXR experience?

**Reply by ZackARC-solutions** [reply]
In my case I add a bunch of glbs to a group and pass that through to MV as a single glb through a blob. Yes, basically an in-world walk atound WebXR experience - we have created a platform to design something and see it in the real world.
